### PR TITLE
Fix some backend problems

### PR DESCRIPTION
**Pull request by zengyan-amazon**
*Issue #, if available:*

*Description of changes:*
* change core.elasticsearch.createClient to core.elasticserach.legacy.createClient to fix breaking change introduced by Kibana master code
* explicitly add auth haader for tenant resolution
* remove test routes
* fix lint issues introduced by SAML code
* fix entity API schema

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: server/routes/index.ts [hunk 1]
```diff
@@ -60,21 +60,24 @@ export function defineRoutes(router: IRouter, esClient: IClusterClient) {
...
   const roleSchema = schema.object({
-    description: schema.string(),
+    description: schema.maybe(schema.string()),
     cluster_permissions: schema.nullable(schema.arrayOf(schema.string())),
     tenant_permissions: schema.arrayOf(schema.any()),
     index_permissions: schema.arrayOf(schema.any()),
   });
 
   const tenantSchema = schema.object({
-    description: schema.string(),
```
Comment (yuxitang-amzn) [inline server/routes/index.ts:77]: For tenant description should be mandatory.

File: server/routes/index.ts [hunk 2]
```diff
@@ -60,21 +60,24 @@ export function defineRoutes(router: IRouter, esClient: IClusterClient) {
...
+    description: schema.maybe(schema.string()),
+    reserved: schema.boolean(),
+    hidden: schema.boolean(),
+    static: schema.boolean(),
```
Comment (yuxitang-amzn) [inline server/routes/index.ts:80]: I don't think reserved, hidden and static is determined by user.

File: server/routes/index.ts [hunk 3]
```diff
@@ -42,15 +42,15 @@ import { API_PREFIX } from '../../common';
 
 export function defineRoutes(router: IRouter, esClient: IClusterClient) {
   const internalUserSchema = schema.object({
-    description: schema.string(),
+    description: schema.maybe(schema.string()),
```
Comment (yuxitang-amzn) [inline server/routes/index.ts:45]: I didn't get the idea for optional `description`. There is no text input for description on the front end page. Is this field defined in the schema of ES index?

File: server/routes/index.ts [hunk 4]
```diff
@@ -42,15 +42,15 @@ import { API_PREFIX } from '../../common';
...
     password: schema.string(),
     backend_roles: schema.arrayOf(schema.string()),
     // opendistro_security_roles: schema.nullable(schema.arrayOf(schema.string())),
     attributes: schema.any(),
   });
 
   const actionGroupSchema = schema.object({
-    description: schema.string(),
+    description: schema.maybe(schema.string()),
```
Comment (yuxitang-amzn) [inline server/routes/index.ts:53]: Same as internal user.

File: server/routes/index.ts [hunk 5]
```diff
@@ -60,21 +60,24 @@ export function defineRoutes(router: IRouter, esClient: IClusterClient) {
   });
 
   const roleMappingSchema = schema.object({
-    description: schema.string(),
+    description: schema.maybe(schema.string()),
```
Comment (yuxitang-amzn) [inline server/routes/index.ts:63]: Same as internal user.

File: server/routes/index.ts [hunk 6]
```diff
@@ -60,21 +60,24 @@ export function defineRoutes(router: IRouter, esClient: IClusterClient) {
...
     backend_roles: schema.arrayOf(schema.string()),
     hosts: schema.arrayOf(schema.string()),
     users: schema.arrayOf(schema.string()),
   });
 
   const roleSchema = schema.object({
-    description: schema.string(),
+    description: schema.maybe(schema.string()),
```
Comment (yuxitang-amzn) [inline server/routes/index.ts:70]: Same as internal user.

File: server/routes/index.ts (hunk 1 above)
Comment (zengyan-amazon) [reply server/routes/index.ts:77]: true, will update

File: server/routes/index.ts (hunk 2 above)
Comment (zengyan-amazon) [reply server/routes/index.ts:80]: agreed

File: server/routes/index.ts (hunk 3 above)
Comment (zengyan-amazon) [reply server/routes/index.ts:45]: hmm, somehow, the legacy API supports this description field, and is saved in ES, so I would prefer to keep this field.

File: server/backend/opendistro_security_client.ts [hunk 7]
```diff
@@ -171,9 +175,9 @@ export class SecurityClient {
           };
         }
         throw Error('failed parsing SAML config');
-      } catch (error) {
-        console.log(error);
-        throw new Error();
+      } catch (parsingError) {
+        console.log(parsingError);
```
Comment (mihirsoni) [inline server/backend/opendistro_security_client.ts:179]: Let's consume kibana's logger so the logs could actually be part of log file.

Comment (zengyan-amazon) [reply server/backend/opendistro_security_client.ts:179]: Yeah, agreed, I plan to change all console.log to Kibana logger. Had `TODO` comments in other places. I will create a task in future sprints to fix this after all auth type implementation is complete

**Pull request merged**